searchlib/src/vespa/searchlib/features/dotproductfeature.cpp

```diff
@@ -271,12 +272,11 @@ DotProductExecutorBase<BaseType>::~DotProductExecutorBase() = default;
 
 template <typename BaseType>
 void DotProductExecutorBase<BaseType>::execute(uint32_t docId) {
-    const AT *values(nullptr);
-    size_t count = getAttributeValues(docId, values);
-    size_t commonRange = std::min(count, _queryVector.size());
+    auto values = getAttributeValues(docId);
+    size_t commonRange = std::min(values.size(), _queryVector.size());
     static_assert(std::is_same<typename AT::ValueType, BaseType>::value);
     outputs().set_number(0, _multiplier.dotProduct(
-            &_queryVector[0], reinterpret_cast<const typename AT::ValueType *>(values), commonRange));
+            &_queryVector[0], reinterpret_cast<const typename AT::ValueType *>(&values[0]), commonRange));
```

Comment:
Consider adding _data()_ to ConstArrayRef to access the raw pointer.

Reply:
Done.